### PR TITLE
Add validation for orientation of initial pose

### LINKED ISSUE #316: Invalid initial pose hangs the node
Invalid initial pose like non-unit quaternion of the orientation hangs the mcl_3dl node.
This occurs when the following invalid initial pose is published.

```
$ rostopic pub -s /initialpose geometry_msgs/PoseWithCovarianceStamped "header:
  seq: 0
  stamp: 'now'
  frame_id: 'map'
pose:
  pose:
    position: {x: 0.0, y: 0.0, z: 0.0}
    orientation: {x: 0.0, y: 0.0, z: 0.0, w: 0.0}  # INVALID
  covariance: [0.01, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.01, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
    0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
    0.0, 0.0, 0.0, 0.0, 0.0, 0.03]" 
```

Adding a validation for the initial pose is required to avoid it.

### PR DESCRIPTION
**Pull request by ykoga-kyutech**
closes #316

### CODE REVIEW

**Comment by at-wat** [comment]
## [[#795](https://travis-ci.org/at-wat/mcl_3dl/builds/691556557)] PASSED on melodic

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_cloud_accum.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_imu_measurement_model_gravity.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_motion_prediction_model_differential_drive.xml: 8 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_noise_generator.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 12 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 8 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_state_6dof.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 10 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-localization_rostest__without_imu_true.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_debug_output_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_landmark_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_incompatible_rostest.xml: 2 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 2 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_debug_output.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 4 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 4 tests
build/test_results/mcl_3dl/rosunit-test_landmark.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat_incompatible.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat_incompatible2.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 2 tests
Summary: 114 tests, 0 errors, 0 failures, 0 skipped
```
</details>

File: src/mcl_3dl.cpp [hunk 1]
```diff
@@ -166,6 +166,17 @@ class MCL3dlNode
...
   void cbPosition(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr& msg)
   {
+    const double len2 =
+        msg->pose.pose.orientation.x * msg->pose.pose.orientation.x +
+        msg->pose.pose.orientation.y * msg->pose.pose.orientation.y +
+        msg->pose.pose.orientation.z * msg->pose.pose.orientation.z +
+        msg->pose.pose.orientation.w * msg->pose.pose.orientation.w;
+    if (std::abs(len2 - 1.0) > 0.1)
+    {
+      ROS_ERROR("Invalid pose is set. The orientation must be unit quaternion.");
```
Comment (at-wat) [inline src/mcl_3dl.cpp:176]: what about 
```suggestion
      ROS_ERROR("Discarded invalid initialpose. The orientation must be unit quaternion.");
```

Comment (ykoga-kyutech) [reply src/mcl_3dl.cpp]: done

**Comment by ykoga-kyutech** [comment]
@at-wat Addressed your comment. PTAL.

The CI failed due to rosdep related problem.

**Comment by at-wat** [comment]
```
/bin/sh: 1: rosdep: not found
```
Official ROS docker image seems broken.

**Comment by at-wat** [comment]
https://github.com/ros-infrastructure/rosdep/issues/745
rosdep seems not installed by default and was installed as a dependency of some package.

**Comment by at-wat** [comment]
## [[#803](https://travis-ci.org/at-wat/mcl_3dl/builds/691586425)] PASSED on melodic

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_cloud_accum.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_imu_measurement_model_gravity.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_motion_prediction_model_differential_drive.xml: 8 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_noise_generator.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 12 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 8 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_state_6dof.xml: 6 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 10 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-localization_rostest__without_imu_true.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_debug_output_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_landmark_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_incompatible_rostest.xml: 2 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 2 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_debug_output.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 4 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 4 tests
build/test_results/mcl_3dl/rosunit-test_landmark.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat_incompatible.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat_incompatible2.xml: 2 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 2 tests
Summary: 114 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Pull request merged**